Source/Core/VideoCommon/CPUCullImpl.h

```diff
@@ -0,0 +1,730 @@
+  o0 = _mm256_broadcast_ps(&vsource[0]);
+  o1 = _mm256_broadcast_ps(&vsource[1]);
+  o2 = _mm256_broadcast_ps(&vsource[2]);
+  o3 = _mm256_setr_ps(0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 1.0f);
+  TransposeYMM(o0, o1, o2, o3);
+}
+
+ATTR_TARGET DOLPHIN_FORCE_INLINE static __m256 ApplyMatrixYMM(__m256 v, __m256 m0, __m256 m1,
+                                                              __m256 m2, __m256 m3)
+{
+  __m256 output = _mm256_mul_ps(_mm256_shuffle_ps(v, v, _MM_SHUFFLE(0, 0, 0, 0)), m0);
+#ifdef USE_AVX2
+  output = _mm256_fmadd_ps(_mm256_shuffle_ps(v, v, _MM_SHUFFLE(1, 1, 1, 1)), m1, output);
+  output = _mm256_fmadd_ps(_mm256_shuffle_ps(v, v, _MM_SHUFFLE(2, 2, 2, 2)), m2, output);
+  output = _mm256_fmadd_ps(_mm256_shuffle_ps(v, v, _MM_SHUFFLE(3, 3, 3, 3)), m3, output);
+#else
+  output = _mm256_add_ps(output, _mm256_mul_ps(_mm256_shuffle_ps(v, v, 0x55), m1));
+  output = _mm256_add_ps(output, _mm256_mul_ps(_mm256_shuffle_ps(v, v, 0xAA), m2));
+  output = _mm256_add_ps(output, _mm256_mul_ps(_mm256_shuffle_ps(v, v, 0xFF), m3));
+#endif
+  return output;
+}
+
+ATTR_TARGET DOLPHIN_FORCE_INLINE static __m256
+TransformVertexNoTransposeYMM(__m256 vertex, __m256 pos0, __m256 pos1, __m256 pos2,  //
+                              __m256 proj0, __m256 proj1, __m256 proj2, __m256 proj3)
+{
+  __m256 mul0 = _mm256_mul_ps(vertex, pos0);
+  __m256 mul1 = _mm256_mul_ps(vertex, pos1);
+  __m256 mul2 = _mm256_mul_ps(vertex, pos2);
+  __m256 mul3 = _mm256_setr_ps(0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 1.0f);
+  __m256 output = _mm256_hadd_ps(_mm256_hadd_ps(mul0, mul1), _mm256_hadd_ps(mul2, mul3));
+  return ApplyMatrixYMM(output, proj0, proj1, proj2, proj3);
+}
+
+template <bool PositionHas3Elems>
+ATTR_TARGET DOLPHIN_FORCE_INLINE static __m256
+TransformVertexYMM(__m256 vertex, __m256 pos0, __m256 pos1, __m256 pos2, __m256 pos3,  //
+                   __m256 proj0, __m256 proj1, __m256 proj2, __m256 proj3)
+{
+  __m256 output = pos3;  // vertex.w is always 1.0
+#ifdef USE_AVX2
+  output = _mm256_fmadd_ps(_mm256_shuffle_ps(vertex, vertex, 0x00), pos0, output);
+  output = _mm256_fmadd_ps(_mm256_shuffle_ps(vertex, vertex, 0x55), pos1, output);
+  if constexpr (PositionHas3Elems)
+    output = _mm256_fmadd_ps(_mm256_shuffle_ps(vertex, vertex, 0xAA), pos2, output);
+#else
+  output = _mm256_add_ps(output, _mm256_mul_ps(_mm256_shuffle_ps(vertex, vertex, 0x00), pos0));
+  output = _mm256_add_ps(output, _mm256_mul_ps(_mm256_shuffle_ps(vertex, vertex, 0x55), pos1));
+  if constexpr (PositionHas3Elems)
+    output = _mm256_add_ps(output, _mm256_mul_ps(_mm256_shuffle_ps(vertex, vertex, 0xAA), pos2));
+#endif
+  return ApplyMatrixYMM(output, proj0, proj1, proj2, proj3);
+}
+
+template <bool PositionHas3Elems, bool PerVertexPosMtx>
+ATTR_TARGET DOLPHIN_FORCE_INLINE static Vector
+Transform2VerticesYMM(const u8* v0data, const u8* v1data,                  //
+                      __m256 pos0, __m256 pos1, __m256 pos2, __m256 pos3,  //
```

Comment:
There are some unused "//" at the end of these lines.

Reply:
They're there to prevent clang-format from reflowing lines